src/dbnode/namespace/kvadmin/ns_admin.go

```diff
@@ -43,38 +44,108 @@ type adminService struct {
 	idGen func() string
 }
 
+const (
+	// M3DBNodeNamespacesKey is the KV key that holds namespaces.
+	M3DBNodeNamespacesKey = "m3db.node.namespaces"
+)
+
 func NewAdminService(store kv.Store, key string, idGen func() string) NamespaceMetadataAdminService {
 	if idGen == nil {
 		idGen = func() string {
 			return uuid.NewV4().String()
 		}
 	}
+	if len(key) == 0 {
+		key = M3DBNodeNamespacesKey
+	}
 	return &adminService{
 		store: store,
 		key:   key,
 		idGen: idGen,
 	}
 }
 
-func (as *adminService) GetAll() ([]*nsproto.NamespaceOptions, error) {
-	// TODO [haijun] move logic from src/query/api/v1/handler/namespace here
-	return nil, ErrNotImplemented
+func (as *adminService) GetAll() (*nsproto.Registry, error) {
```

Comment:
This is cool. Are we gonna replace the handler code with the service soon? Do we have an open issue for that?

Reply:
let me open an issue to track it. I won't get to it before my vacation... so it won't be soon